File: SocketComm.Examples/SendArray/OneDClient.cs
using static SocketComm.Cmd;

namespace SocketComm.Examples.SendArray;

public class OneDClient
{
    public static void Main()
    {
        FPipe p = new FPipe("/tmp/OneDArray.pipe");
        p.Connect();
        p.WriteCmd(Ready);
        Cmd c;
        c = p.ReadCmd();
        while (c == Receive)
        {
            int[] i =(int[]) p.ReceiveValue((val) => SimpleConversion.BytesToArray(val, SimpleConversion.BytesToNumber<int>));
            Console.Write("Received [");
            foreach (var e in i)
            {
                Console.Write(e + ", ");
            }
            Console.WriteLine("]");
            p.WriteCmd(Ready);
            p.ExpectCmd(Go);
            var res = Bench(i);
            p.WriteCmd(Done);
            Console.WriteLine($"Sending {res}");
            p.SendValue(res ,SimpleConversion.NumberToBytes);
            c = p.ReadCmd();
        }
        p.Dispose();
    }

    public static int Bench(int[] i)
    {
        var sum = 0;
        foreach (var e in i)
        {
            sum += e;
        }

        return sum;
    }
}

File: SocketComm.Examples/SendArray/OneDClient.java
import java.io.Console;
import java.nio.ByteBuffer;

public class OneDClient {
    public static void main(String[] args) throws Exception {
        FPipe p = new FPipe("/tmp/OneDArray.pipe");
        p.WriteCmd(Cmd.Ready);
        Cmd c;
        c = p.ReadCmd();
        while (c == Cmd.Receive)
        {
            var i = (Integer[])p.ReceiveValue((buf)-> SimpleConversion.BytesToArray(buf, ByteBuffer::getInt));
            System.out.print("Received [");
            for (var e : i) {
                System.out.print(e +", ");
            }
            System.out.println("]");
            p.WriteCmd(Cmd.Ready);
            p.ExpectCmd(Cmd.Go);
            var res = Bench(i);
            p.WriteCmd(Cmd.Done);
            System.out.printf("Sending %d\n", res);
            p.SendValue(res, (val)-> ByteBuffer.allocate(4).putInt(val));
            c = p.ReadCmd();
        }
        p.close();
    }

    public static int Bench(Integer[] i){

        var sum = 0;
        for (var e : i){
            sum += e;
        }
        return sum;
    }
}
